clients/drcachesim/analyzer.cpp

```diff
@@ -295,6 +327,16 @@ analyzer_tmpl_t<RecordType, ReaderType>::init_scheduler_common(
     std::vector<typename sched_type_t::input_workload_t> &workloads,
     typename sched_type_t::scheduler_options_t options)
 {
+    // TODO i#7216: temporary default metadata info for the noise generator.
+    // We plan to set these by analyzing the workloads.
+    options.noise_generator_info =
+        std::unique_ptr<noise_generator_info_t>(new noise_generator_info_t);
```

Comment:
Not seeing why this structure lives in the scheduler options: it seems it should just be an input to a noise generator-owned factory for creating noise workloads.

Reply:
Right, removed now that we have a noise generator factory.